### PR TITLE
accept a function for sourceMapURL

### PR DESCRIPTION
**Pull request by montmanu**

### CODE REVIEW

**Comment by XhmikosR** [comment]
@montmanu: can you rebase?

File: tasks/less.js [hunk 1]
```diff
@@ -126,7 +126,15 @@ module.exports = function(grunt) {
...
-    if (typeof options.sourceMap === 'boolean' && options.sourceMap) {
+    if (_.isFunction(options.sourceMapURL)) {
+      try {
+        options.sourceMapURL = options.sourceMapURL(destFile);
+      } catch (e) {
+        grunt.fail.warn(wrapError(e, 'Generating sourceMapURL failed.'));
+      }
+    }
+
+    if (_.isBoolean(options.sourceMap) && options.sourceMap) {
```
Comment (XhmikosR) [inline tasks/less.js]: `typeof options.sourceMap === 'boolean'`

File: tasks/less.js [hunk 2]
```diff
@@ -126,6 +126,14 @@ module.exports = function(grunt) {
       }
     }
 
+    if (_.isFunction(options.sourceMapURL)) {
```
Comment (XhmikosR) [inline tasks/less.js]: Isn't `typeof  options.sourceMapURL === 'function'` enough here?

Comment (montmanu) [inline tasks/less.js]: probably. just figured since lodash had already been imported, would use that utility. would you prefer `typeof` ?

Comment (XhmikosR) [inline tasks/less.js]: I was just trying to decouple of lodash and just keep stuff we only need.

Comment (montmanu) [inline tasks/less.js]: ok, cool. will update with `typeof` here and in the other PR.

**Pull request merged**